test/widgets/docs_screenshots/conftest.py

```diff
@@ -0,0 +1,122 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import json
+import os
+import shutil
+from pathlib import Path
+
+import pytest
+
+from libqtile.bar import Bar
+from libqtile.config import Group, Screen
+
+
+@pytest.fixture(scope="session")
+def target():
+    folder = Path(__file__).parent / "screenshots"
+    docs_folder = Path(__file__).parent / ".." / ".." / ".." / "docs" / "_static" / "screenshots" / "widgets"
+    log = os.path.join(docs_folder, "shots.json")
+    if os.path.isdir(folder):
+        shutil.rmtree(folder)
+    os.makedirs(folder)
+    key = {}
+
+    def get_file_name(w_name, config):
+        nonlocal key
+
+        # Convert config into a string of key=value
+        entry = ", ".join(f"{k}={repr(v)}" for k, v in config.items())
+
+        # Check if widget is in the key dict
+        if w_name not in key:
+            key[w_name] = {}
+
+        # Increment the index number
+        indexes = [int(x) for x in key[w_name]]
+        index = max(indexes) + 1 if indexes else 1
+
+        # Record the config
+        key[w_name][index] = entry
+
+        # Define the target folder and check it exists
+        shots_dir = os.path.join(folder, w_name)
+        if not os.path.isdir(shots_dir):
+            os.mkdir(shots_dir)
+
+        # Returnt the path for the screenshot
+        return os.path.join(shots_dir, f"{index}.png")
+
+    yield get_file_name
+
+    # We copy the screenshots from the test folder to the docs folder at the end
+    # This prevents pytest deleting the files itself
+
+    # Remove old screenshots
+    if os.path.isdir(docs_folder):
+        shutil.rmtree(docs_folder)
+
+    # Copy to the docs folder
+    shutil.copytree(folder, docs_folder)
+    with open(log, "w") as f:
+        json.dump(key, f)
+
+    # Clear up the tests folder
+    shutil.rmtree(folder)
+
+
+@pytest.fixture
+def screenshot_manager(widget, request, manager_nospawn, minimal_conf_noscreen, target):
```

Comment:
The `widget` fixture is defined in each of the individual `ss_widgetname.py` file.